### ISSUE TITLE
Is it actually returning 0 on errors?

### ISSUE THREAD

**Issue by TeaWhyDee** [flyingrub/scdl#384]
Since scdl errors out 90% of the time, I made a shell script to retry until it succeeds, but scdl always succeeds (returns 0 on exit) even if there was an error!

Script: 
`until scdl -l <link> -c; do echo restarting; done`

Similar script was working with other programs.
So instead I just run it in a while loop, but then there is no way to know when it succeeds :(

**Comment by def-hash** [comment]
Check out my pull request, you can see the changes I made to add the user agent to the request headers. If you do it, the program works as expected without retrying.

**Comment by def-hash** [comment]
> Since scdl errors out 90% of the time, I made a shell script to retry until it succeeds, but scdl always succeeds (returns 0 on exit) even if there was an error!
> 
> Script: `until scdl -l <link> -c; do echo restarting; done`
> 
> Similar script was working with other programs. So instead I just run it in a while loop, but then there is no way to know when it succeeds :(

Check out my pull request, you can see the changes I made to add the user agent to the request headers. If you do it, the program works as expected without retrying.

**Comment by TeaWhyDee** [comment]
Thank you, your branch works fine. But I think this issue should still be fixed.

**Comment by 7x11x13** [comment]
Could you provide a test case where the script fails and returns an exit code of 0?

**Comment by TeaWhyDee** [comment]
for example:
```
$ scdl -l https://soundcloud.com/link_that_doesnt_exist && echo "Command returned 0"
Soundcloud Downloader
Error resolving url, retrying...
Failed to get item...
Command returned 0
```
from the output we can see that it returns 0. I tried the same command with other python scripts and echo doesn't run if they error.

**Comment by 7x11x13** [comment]
Thanks, I believe I fixed this in the latest version (2.2.5)
